server/src/signal.rs

```diff
@@ -73,6 +72,20 @@ pub(super) async fn sighusr2_compact() {
     }
 }
 
+pub(super) async fn sigusr2_tokio_dump() {
+    let mut stream =
+        signal(SignalKind::user_defined2()).expect("failed to register handler for SIGHUP");
+
+    let tc = restate_core::TaskCenter::current();
+
+    loop {
+        stream.recv().await;
+        warn!("Received SIGUSR2, dumping tokio task backtraces");
+
+        tc.dump_tasks(std::io::stderr()).await;
```

Comment:
Should we spawn this in a task? I'm concerned that this might not be cancellation safe, or if it is, that assumption can change at any time.